src/com/facebook/buck/parser/UnexpectedFlavorException.java

```diff
@@ -114,13 +116,24 @@ public static UnexpectedFlavorException createWithSuggestions(
     return builder.build();
   }
 
+  private static ImmutableSet<Flavor> getValidFlavors(Flavored flavored, BuildTarget target) {
+    ImmutableSet.Builder<Flavor> builder = ImmutableSet.builder();
+    for (Flavor flavor : target.getFlavors()) {
+      if (flavored.hasFlavors(ImmutableSet.of(flavor))) {
+        builder.add(flavor);
+      }
+    }
+    return builder.build();
+  }
+
   private static String createDefaultMessage(
-      BuildTarget target, ImmutableSet<Flavor> invalidFlavors) {
+      BuildTarget target, ImmutableSet<Flavor> invalidFlavors, ImmutableSet<Flavor> validFlavors) {
     ImmutableSet<String> invalidFlavorsStr =
         invalidFlavors.stream().map(Flavor::toString).collect(MoreCollectors.toImmutableSet());
     String invalidFlavorsDisplayStr = String.join(", ", invalidFlavorsStr);
     return String.format(
-        "The following flavor(s) are not supported on target %s:\n%s.",
-        target, invalidFlavorsDisplayStr);
+        "The following flavor(s) are not supported on target %s:\n%s. Available flavors are: %s",
+        target, invalidFlavorsDisplayStr, validFlavors.stream().map(Flavor::getName).collect(
```

Comment:
nit: please extract the last expression into a local variable, since it's not on the same semantical level as the other argument statements